### PR TITLE
Move @microsoft/api-extractor to web-build-tools

### PR DESCRIPTION
**Pull request by nickpape-msft**
A few things were updated:

* Copied over api-extractor
  * Typings for z-schema were slightly different. Switched to using @types.
  * Small conflict wrt typescript types, should be resolved in a later PR which makes tslint, typescript, API extractor all use the same compiler options.
  * Changed build rig for API-Extractor to `@microsoft/node-library-build`
  * Updated LICENSE to MIT
* Updated dependencies for `gulp-core-build-typescript`, copied over `ApiExtractorTask`

These files were taken from sp-client repository - the SHA1 of the commit they are based on is:

`97f7928b0d64140b8940c7605d47f3d45c7ad5fb`

### CODE REVIEW

File: common/reviews/PackageDependencies.json [hunk 1]
```diff
@@ -44,8 +48,12 @@
       "allowedCategories": [ "libraries" ]
     },
     {
+      "name": "@microsoft/sp-tslint-rules",
```
Comment (iclanton) [inline common/reviews/PackageDependencies.json]: This should get removed too.

File: api-extractor/api-extractor.csproj [hunk 2]
```diff
@@ -0,0 +1,80 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (pgonzal) [inline api-extractor/api-extractor.csproj]: Agreed

File: api-extractor/gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+const build = require('@microsoft/node-library-build');
+
+const tslintCommon = require('@microsoft/sp-tslint-rules');
```
Comment (pgonzal) [inline api-extractor/gulpfile.js]: Looking at tslint.json:

```json
  "rules": {
    "prefer-const": true,
    "react-a11y-aria-unsupported-elements": true,
    "react-a11y-event-has-role": true,
    "react-a11y-image-button-has-alt": true,
    "react-a11y-img-has-alt": true,
    "react-a11y-lang": true,
    "react-a11y-anchors": false,
    "react-a11y-meta": true,
    "react-a11y-props": true,
    "react-a11y-proptypes": true,
    "react-a11y-role-has-required-aria-props": true,
    "react-a11y-role-supports-aria-props": true,
    "react-a11y-role": true,
    "react-a11y-tabindex-no-positive": true,
    "react-a11y-titles": true
  }
```

Is prefer-const not enabled in GCB?  That is a pretty fundamental rule -- can we enable it somehow?

File: api-extractor/LICENSE [hunk 4]
```diff
@@ -0,0 +1,24 @@
+@microsoft/gulp-core-build
```
Comment (pgonzal) [inline api-extractor/LICENSE]: This is the wrong package name

File: api-extractor/README.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# api-extractor
```
Comment (pgonzal) [inline api-extractor/README.md]: Also update the top-level README

**Comment by pgonzal** [comment]
Don't squash this change; I'd like to preserve your ffe62f340a6c91c166cf421ab02482738c696018 as close as possible to the upstream branch.

File: api-extractor/README.md (hunk 5 above)
Comment (nickpape-msft) [inline api-extractor/README.md]: Fixed

File: api-extractor/LICENSE (hunk 4 above)
Comment (nickpape-msft) [inline api-extractor/LICENSE]: Fixed

File: api-extractor/gulpfile.js (hunk 3 above)
Comment (nickpape-msft) [inline api-extractor/gulpfile.js]: Not sure where you are seeing that lint file. The default lint rules that actually are used by `node-library-build` are in `gulp-core-build-typescript/src/defaultTsLint.json`

I enabled it there, but for now this package has a cyclic dependency and therefore it builds using the last published version of the repo, so the rule will be disable until we republish.

File: common/reviews/PackageDependencies.json (hunk 1 above)
Comment (nickpape-msft) [inline common/reviews/PackageDependencies.json]: I will remove this dependency

File: api-extractor/api-extractor.sln [hunk 6]
```diff
@@ -0,0 +1,19 @@
+﻿
```
Comment (nickpape-msft) [inline api-extractor/api-extractor.sln]: Removed

**Pull request merged**